File: protobuf-capnproto-and-zeromq/zmq/subscriber.cpp
#include <zmqpp/zmqpp.hpp>
#include <string>
#include <iostream>

using namespace std;

int main(void) {
    const string endpoint = "tcp://localhost:4242";

    zmqpp::context context;
    zmqpp::socket_type type = zmqpp::socket_type::subscribe;
    zmqpp::socket socket (context, type);

    socket.subscribe(""); // subscribe the default channel
    
    zmqpp::message_t message;
    cout << "Connecting to endpoint: " << endpoint << "...\n";
    socket.connect(endpoint);
    cout << "Endpoint connected" << endl;
    while (true) {
        socket.receive(message);
        cout << message.get(0)
             << " (size: " << message.size(0) << " bytes)" << endl;
    }
    return 0;
}

File: zeromq/subscriber.py
import logging
import zmq

context = zmq.Context()

#  Socket to talk to server
logging.info("Connecting to publisher endpoint")
socket = context.socket(zmq.SUB)
socket.connect("tcp://127.0.0.1:4242")
logging.info("Connected to endpoint")


while True:
    #  Get the reply.
    message = socket.recv()
    print(f"Image received [{len(message)}]")
    with open("/tmp/test.jpg", "wb") as binary_file:
        # Write bytes to file
        binary_file.write(message.data())